### DISCUSSION TITLE
Can I use NeuroKit to process a list of RR intervals?

### DISCUSSION THREAD

**Discussion by apar945** [neuropsychology/NeuroKit#819] [Q&A] [answered]
Hello,

I have reviewed the functions relevant to HRV analysis and PPG data analysis. I have a dataset that contains a list of RR intervals from the ECG sensor and similar list of beat to beat intervals from PPG sensors. In short the work to extract the intervals in milliseconds is done already.

I would like to know if I can use the Neurokit library to process this list to generate time and frequency domain HRV analysis?
Note that I do not have raw ECG or PPG waveform data so I am unable to provide additional information required by these functions.

I was trying to look for ectopic beat detection and correction methods and couldn't find them in the library too.

I would like to use the library for data analysis if possible for the data collected as part of my PhD study.

Regards
Ankit

**Comment by DominiqueMakowski** [comment, marked as answer]
You can convert your list of RR intervals using [**nk.intervals_to_peaks()**](https://neuropsychology.github.io/NeuroKit/functions/hrv.html#intervals-to-peaks) to recover raw R-peaks locations and then use all the neurokit functionalities ☺️

**Comment by apar945** [comment]
Hello Dominique,

Thank you for the highlighting this function. I have a follow up question.

Are there any function for artifact correction, ectopic beats removal and outlier removal (RR intervals less than 400ms or above 2000ms)?
